SwiftWisdomTests/StandardLibrary/Numbers/CountableClosedRange+ExtensionsTests.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  CountableClosedRange+ExtensionsTests.swift
+//  SwiftWisdom
+//
+//  Created by Andrew Dolce on 11/21/16.
+//  Copyright © 2016 Intrepid. All rights reserved.
+//
+
+import XCTest
+@testable import SwiftWisdom
+
+class CountableClosedRangeExtensionTests: XCTestCase {
+
+    // This test is complex. Here is an explanation as to why:
+    //
+    // The original goal was to test the public ip_random() function on CountableClosedRange. Previously we had tests
+    // that would call ip_random() a few times, and then verify that the output values were contained inside the
+    // original range as expected.
+    //
+    // I've decided to change this approach because:
+    //
+    // 1. Testing the output of real calls to arc4random_uniform means that the unit test will be non-deterministic, and
+    //    therefore could be flaky. What if we had a bug that only occurred with a single specific value of arc4random?
+    //    Our test would pass most of the time, but occasionally fail.
+    //
+    // 2. We're not really interested in testing the behavior of arc4random in the first place, but rather our logic
+    //    that sits on top of that function.
+    //
+    // One approach would be to seed the random number generator with a hard-coded value, but arc4random is not designed
+    // to be seedable.
+    //
+    // Instead I decided to break out an internal version of ip_random that allows the caller to specify a replacement
+    // for arc4random_uniform. This way we can mock out the function to return controlled values. The obvious downside
+    // is that the test is now somewhat coupled to the implementation, and tests an internal helper instead of the true
+    // public interface.
+
+    func testRandomInRange() {
+        let lowerBound = -100
+        let upperBound = 100
+        let testRange = lowerBound...upperBound
+
+        // We do not want to assume that we know the upper bound argument provided to the random number generator
+        // function. Instead we will make one call and capture the input.
+        var capturedRNGUpperBound: Int? = nil
+        let _ = testRange.ip_random(withUniformRNGFunction: { upperBound in
+            capturedRNGUpperBound = upperBound
+            return 0
+        })
+
+        guard let rngUpperBound = capturedRNGUpperBound, rngUpperBound > 0 else {
+            XCTAssert(false, "Expected that RNG function upper bound \(capturedRNGUpperBound) is non-nil and positive.")
```

Comment:
Also, per guard style guide, we should not use them as control flow, but only for early return. In addition, i dont think if lets are better in tests.